packages/graphql/src/types/app.ts

```diff
@@ -66,65 +87,67 @@ export type QuestionOptionsNumerical = {
   }[]
 }
 
-export type QuestionOptionsFreeText = {
+export interface ElementOptionsFreeText extends BaseElementOptions {
   solutions?: string[]
   restrictions?: {
     maxLength?: number | null
   }
 }
 
-export type QuestionOptions =
-  | QuestionOptionsChoices
-  | QuestionOptionsNumerical
-  | QuestionOptionsFreeText
+export type ElementOptions =
+  | ElementOptionsChoices
+  | ElementOptionsNumerical
+  | ElementOptionsFreeText
+
+export interface BaseElementData {
+  type: ElementType
 
-export interface BaseQuestionData {
   id: number
   name: string
-  type: QuestionType
-  displayMode: QuestionDisplayMode
   content: string
   pointsMultiplier: number
-  explanation: string | null
+  explanation?: string | null
+
+  options: object
+
+  // TODO: these legacy props have been moved to options
```

Comment:
So we would have to change the `options: object` above to `options: ElementOptions` and save the corresponding values at the correct place during the creation of instances?

Reply:
no, in base element data (which is used only to extend from), we do not care what is in the options. that will be specified in the type specific interfaces that extend from this here :)

Reply:
in terms of db maintenance, we will need to move these props into the options object, that is correct, but that does not change anything here except that the three properties will disappear from this type and be added to the specific options type.

Reply:
makes sense, thanks for the explanation :)